dandi/download.py

```diff
@@ -1143,17 +1161,9 @@ def _check_if_more_attempts_allowed(
                 sleep_amount,
                 exc,
             )
-        else:
-            lgr.debug(
-                "%s - download failed on attempt #%d: %s, will sleep a bit and retry",
-                path,
-                attempt,
-                exc,
-            )
-    # if is_access_denied(exc) or attempt >= 2:
-    #     raise
-    # sleep a little and retry
-    else:
+    if sleep_amount < 0:
+        # it was not Retry-after set, so we come up with random duration to sleep
```

Comment:
In the current arrangement, the execution flow can reach here event if `Retry-After` is set. It can be set to an extremely close datetime or some string can't be parsed properly in to a `datetime` object